### PR TITLE
Fix docker compose not being able to stat files

### LINKED ISSUE #15941: Docker compose cp: cannot stat '/tmp/data-ps/prestashop/*'
docker-compose up 

**error message**
Reapplying PrestaShop files for enabled volumes ...
cp: cannot stat '/tmp/data-ps/prestashop/*': No such file or directory

**To Reproduce**
1. chown -hR www-data:www-data ./prestashop
2. git clone https://github.com/PrestaShop/PrestaShop.git
3. cd prestashop
4. docker-compose up

**Additional information**
docker
Docker version 19.03.2, build 6a30dfc
docker-compose
docker-compose version 1.24.1, build 4667896b

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Be sure we use the primary user (1000 in Unix environment)
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #15941
| How to test?  | docker-compose up --build<br> Navigate to http://localhost:8001

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15942)

### CODE REVIEW

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
Need one more review

@Quetzacoalt91 ?

File: docker-compose.yml [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
-            DB_SERVER: mysql
-            PS_DOMAIN: localhost:8001
-            PS_FOLDER_INSTALL: install-dev
-            PS_FOLDER_ADMIN: admin-dev
-        ports:
-            - "8001:80"
-        volumes:
-            - ./:/var/www/html/:delegated
-            - vendor:/var/www/html/vendor
-            - var:/var/www/html/var
```
Comment (Quetzacoalt91) [inline docker-compose.yml:35]: These volumes are useful for people running Docker on Mac.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,37 +1,31 @@
...
+      DB_NAME: prestashop
+      DB_SERVER: mysql
+      PS_DOMAIN: localhost:8001
+      PS_FOLDER_INSTALL: install-dev
+      PS_FOLDER_ADMIN: admin-dev
+    ports:
+      - "8001:80"
+    volumes:
+      - ./:/var/www/html/
+    restart: always
```
Comment (Quetzacoalt91) [inline docker-compose.yml]: `restart:always` is actually unecessary.We don't aim production environments with this stack so we do not need to start it automatically.

**Review by matks — DISMISSED** [review]

**Comment by boubkerbribri** [comment]
Error when running command `docker-compose up`

image

**Comment by PierreRambaud** [comment]
No more in WIP.
Can a developer check if everything is ok :)

File: .docker/docker_run_git.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+if [ ! -f ./config/settings.inc.php ]; then
+    if [ $PS_INSTALL_AUTO = 1 ]; then
+
+        echo "\n* Installing PrestaShop, this may take a while ...";
+
+        if [ $PS_ERASE_DB = 1 ]; then
+            echo "\n* Drop & recreate mysql database...";
+            if [ $DB_PASSWD = "" ]; then
+                echo "\n* Dropping existing database $DB_NAME..."
+                mysql -h $DB_SERVER -P $DB_PORT -u $DB_USER -p$DB_PASSWD -e "drop database if exists $DB_NAME;"
```
Comment (Progi1984) [inline .docker/docker_run_git.sh]: @PierreRambaud The drop command is identical in the `if` and the `else`. You can share it before the `if`.

**Review by Progi1984 — COMMENTED** [review]
A small change from my side. But I ask a second review (@PrestaShop/prestashop-core-developers).

File: .docker/docker_run_git.sh (hunk 3 above)
Comment (PierreRambaud) [reply .docker/docker_run_git.sh]: the password was empty and the `-p$DB_PASSWD` shouldn't be used

**Review by Progi1984 — APPROVED** [review]
Need a second review

File: .gitignore [hunk 4]
```diff
@@ -97,6 +97,7 @@ themes/*/cache/*
 
 # Config
 
+config/defines_custom.inc.php
```
Comment (jolelievre) [inline .gitignore:100]: What's the relation with this Docker PR?

File: docker-compose.yml [hunk 5]
```diff
@@ -1,37 +1,35 @@
...
+    build: .docker
+    environment:
+      PS_INSTALL_AUTO: 1
+      DB_PASSWD: prestashop
+      DB_NAME: prestashop
+      DB_SERVER: mysql
+      PS_DOMAIN: localhost:8001
+      PS_FOLDER_INSTALL: install-dev
+      PS_FOLDER_ADMIN: admin-dev
+    command: ["/tmp/wait-for-it.sh", "--timeout=60", "--strict", "mysql:3306", "--", "/tmp/docker_run_git.sh"]
```
Comment (jolelievre) [inline docker-compose.yml:29]: So you need to run this command in both `docker-compose.yml` and in `.docker/Dockerfile` ?

**Review by jolelievre — COMMENTED** [review]
Just two questions

File: .gitignore (hunk 4 above)
Comment (PierreRambaud) [reply .gitignore:100]: this file is created while running the container or in some situation https://github.com/PrestaShop/PrestaShop/blob/develop/config/config.inc.php#L30
but didn't know why it wasn't be ignored :thinking:

File: docker-compose.yml (hunk 5 above)
Comment (PierreRambaud) [reply docker-compose.yml:29]: Dockerfile is not related to the docker-compose behavior.
In `.docker/Dockerfile` we are building the container architecture / behavior, whereas when you use the docker-compose, you are waiting for some service to be up

**Review by jolelievre — APPROVED** [review]
Thanks @PierreRambaud

**Comment by jolelievre** [comment]
Tested and approved!

**Pull request merged**